### PR TITLE
add proxy config to golang sdk

### PR DESCRIPTION
**Pull request by RazcoDev**

### CODE REVIEW

File: pkg/api/proxyConfig.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+func (p *ProxyConfigs) List(ctx context.Context, page int, perPage int) ([]models.ProxyConfigRead, error) {
+	perPageLimit := int32(DefaultPerPageLimit)
+	if !isPaginationInLimit(int32(page), int32(perPage), perPageLimit) {
+		err := errors.NewPermitPaginationError()
+		p.logger.Error("error listing ProxyConfigs - max per page: "+string(perPageLimit), zap.Error(err))
+		return nil, err
+	}
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
+	if err != nil {
+		p.logger.Error("", zap.Error(err))
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:39]: empty error message ?

File: pkg/api/proxyConfig.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+// Usage Example:
+// `ProxyConfigs, err := PermitClient.Api.ProxyConfigs.List(ctx, 1, 10)`
+func (p *ProxyConfigs) List(ctx context.Context, page int, perPage int) ([]models.ProxyConfigRead, error) {
+	perPageLimit := int32(DefaultPerPageLimit)
+	if !isPaginationInLimit(int32(page), int32(perPage), perPageLimit) {
+		err := errors.NewPermitPaginationError()
+		p.logger.Error("error listing ProxyConfigs - max per page: "+string(perPageLimit), zap.Error(err))
+		return nil, err
+	}
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:37]: it is stated that it required project level api key in line 27 but here it requires environment level api key

File: pkg/api/proxyConfig.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	return ProxyConfigs, nil
+}
+
+// Get a Proxy Config by key, requires Environment level API key, or higher.
+// Usage Example:
+// `proxyConfig, err := PermitClient.Api.ProxyConfigs.Get(ctx, "proxyconfig-key")`
+func (p *ProxyConfigs) Get(ctx context.Context, proxyConfigKey string) (*models.ProxyConfigRead, error) {
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
+	if err != nil {
+		p.logger.Error("", zap.Error(err))
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:58]: empty error message ?

File: pkg/api/proxyConfig.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	if err != nil {
+		p.logger.Error("error getting Proxy Config: "+proxyConfigKey, zap.Error(err))
+		return nil, err
+	}
+	return proxyConfig, nil
+}
+
+// GetByKey get a project by key, requires Project level API key, or higher.
+// Usage Example:
+// `project, err := PermitClient.Api.ProxyConfigs.GetByKey(ctx, "project-key")`
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:72]: project ?

File: pkg/api/proxyConfig.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+// Usage Example:
+// ```
+// proxyConfigUpdate := models.NewProxyConfigUpdate()
+// proxyConfigUpdate.SetName("new-proxy-config-name")
+// proxyConfig, err := PermitClient.Api.ProxyConfigs.Update(ctx, "proxy-config-key", proxyConfigUpdate)
+// ```
+func (p *ProxyConfigs) Update(ctx context.Context, proxyConfigKey string, proxyConfigUpdate models.ProxyConfigUpdate) (*models.ProxyConfigRead, error) {
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
+	if err != nil {
+		p.logger.Error("", zap.Error(err))
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:115]: empty error message ?

File: pkg/api/proxyConfig.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+	return proxyConfig, nil
+}
+
+// Delete a proxy config, requires Environment level API key, or higher.
+// Usage Example:
+// `err := PermitClient.Api.ProxyConfigs.Delete(ctx, "proxy-config-key")`
+func (p *ProxyConfigs) Delete(ctx context.Context, proxyConfigKey string) error {
+	err := p.lazyLoadPermitContext(ctx, config.EnvironmentAPIKeyLevel)
+	if err != nil {
+		p.logger.Error("", zap.Error(err))
```
Comment (omer9564) [inline pkg/api/proxyConfig.go:133]: empty error message ?

**Review by omer9564 — CHANGES_REQUESTED** [review]

File: pkg/api/proxyConfig.go (hunk 1 above)
Comment (RazcoDev) [reply pkg/api/proxyConfig.go:39]: The `lazyLoadPermitContext` error message is enough

Comment (RazcoDev) [reply pkg/api/proxyConfig.go:39]: It's like this across the SDK

File: pkg/api/proxyConfig.go (hunk 4 above)
Comment (RazcoDev) [reply pkg/api/proxyConfig.go:72]: fixed

File: pkg/api/proxyConfig.go (hunk 3 above)
Comment (RazcoDev) [reply pkg/api/proxyConfig.go:58]: same as above

File: pkg/api/proxyConfig.go (hunk 5 above)
Comment (RazcoDev) [reply pkg/api/proxyConfig.go:115]: Same as above

File: pkg/api/proxyConfig.go (hunk 6 above)
Comment (RazcoDev) [reply pkg/api/proxyConfig.go:133]: same as above

**Review by orishavit — APPROVED** [review]

**Pull request merged**